recipes/range-v3/all/CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(cmake_wrapper)
+
+include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
```

Comment:
I still don't see why this would be necessary for a header-only library.
I understand that you are reusing the ``cmake.install()`` functionality to create the final package, but still reads like a very complex recipe to understand for a header-only library.

Reply:
I think this one is an more or less an official practice, isn't it?
e.g. it's used by our [templates](https://github.com/bincrafters/templates/blob/master/default/CMakeLists.txt), recommended by [getting started guide](https://docs.conan.io/en/latest/creating_packages/getting_started.html) and generated by [conan new](https://docs.conan.io/en/latest/reference/commands/creator/new.html) command.
and there is nothing complex, just include a file produced by generator, right?

Reply:
I disagree:
- The getting started guide does not suggest such wrapping CMakeLists.txt. 
- The ``conan new -i`` for header only generates a simple ``conanfile.py`` with just a package() method, not such structure.

I really think we should aim for header-only library to something that simple that authors will package that without even thinking. Having a wrapping CMakeLists.txt that includes a generated conanbuildinfo.cmake file, that is later used in a build() method to call the ``cmake.install()`` functionality is totally overkill for 99% of the cases, and something that only savvy Conan users will understand when reading.

Reply:
Actually, our header_only template doesn't use CMake at all: https://github.com/bincrafters/templates/tree/master/header_only

That being said, I think that when the upstream project uses CMake and offers an installation step, that we should totally use it, instead of re-implementing this logic

The CMake wrapper is almost always the same identical boiler plate. So it is true, that people who want to understand these recipes need to get the idea what this wrapper is doing, but pretty much just one time, afterwards it stays the same

Reply:
Then, where the exported wrapping CMakeLists.txt comes from? It will improve dropping it, that is possibly the most weird thing and difficult to understand. Having everything in the conanfile.py, even if calling cmake.install() is way easier to understand than having such added CMakeLists.txt

Reply:
previously, we used injection of cmake code with `tools.replace_in_file`. however, this pattern was proven to be fragile, as libraries tend to change theirs `CMakeLists.txt`, so `replace_in_file` stopped to find required pattern to replace. so, exporting wrapping `CMakeLists.txt` was considered as an alternative.
however, it suffers from its own problems. one of them - many projects are not ready/tested from being included as sub-directory, e.g. they use paths relative to `CMAKE_SOURCE_DIR`/`CMAKE_BINARY_DIR` instead of `CMAKE_CURRENT_SOURCE_DIR`/`CMAKE_CURRENT_BINARY_DIR`.
therefore, both solutions are ugly and fragile right now.
right now, `conan new` (without `-i`) and conan getting starting guide use the first method for injection, while bincrafters templates use the second.
however, my comment wasn't about the injection method, but about the injecting code - `include(conanbuildinfo.cmake)`. it's used by all approaches right now, and this injection appears to be required by many reasons, as:
- it provides conan generated variables and targets
- it sets some compiler flags to ensure consistency with conan settins
- it sets up output directories to ensure the correct layout
- it does many other things (checking compiler, setting up rpath, etc.)
and in order to get rid of such thing, and use the transparent integration, I suppose we should switch to the CMake toolchain approach, which will be used by CMake build helper.